### ISSUE TITLE
Extend git command for convenient.

### ISSUE THREAD

**Issue by romandev** [lunchclass/absolute#257]
For example, if the following command runs, push the local change into remote branch and then make a pull request automatically.
`$ git cl upload`

Another example, if the following command runs, we can cherry-pick the related pull request.
`$ git cl patch {#pull_request_number}`
`$ git cl patch #123`

Reference Link: [Chromium is using |depot_tools| for this.](http://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html)

**Comment by nadongguri** [comment]
https://hub.github.com/

**Comment by rockplani** [comment]
Hello. I want to participate in this part with my bro for the absoulte project. 
And i hope to be included in 'Assignees' of this issue. 
But i have no authority about assigning self. 
So can you include me in your collaborator?

**Comment by rockplani** [comment]
Greetings again. I need advice from romandev on this issue.
As you know, git can create a custom command for convenient by setting an alias in .config.

[Alias]
Cma = git add. && git commit -m 'cma'

$ Git cma

For a pull-request (called 'pr'), you can control the existing pr on the command line. The following open source links are examples. This allows [git cl patch #pr_num] for that issue.
(https://gist.github.com/gnarf/5406589) 

However, if you are creating a pr, you need a procedure on the web. In other words, commands that were not originally available in cmd will require another attempt.

So we looked at several open sources that enable 'make pr' in cmd.

Https://github.com/robinsmidsrod/App-GitHubPullRequest
Https://hub.github.com/

These have improved the 'alias' by interfacing with the bash script inside the project, and eventually enabled web-based operations on the command line.

As above, I want to know if I can put the script in 'bootstrap / command / git' and support the necessary functions. In addition, [git commit - git cl upload] in chromium and [git push origin branch - pull request
In github web] are the same? ask politely. 

Thank you.

**Comment by romandev** [comment]
Hi @rockplani,
Thank you for your contribution.
I left some inline comments.

> However, if you are creating a pr, you need a procedure on the web. In other words, commands that were not originally available in cmd will require another attempt.

What procedure do we need? I didn't understand your intention exactly but
I guess it will be like that:
- We will need the following commands.
  - `git cl patch` for cherry-pick patches from a pull request.
    - example: https://gist.github.com/gnarf/5406589
  - `git cl upload` for making a new pull request on the command line.
    - one example: https://github.com/robinsmidsrod/App-GitHubPullRequest
    - another example: https://hub.github.com/

Is my understanding correct?

> These have improved the 'alias' by interfacing with the bash script inside the project, and eventually enabled web-based operations on the command line.

Yeah, you're right.

> 
> As above, I want to know if I can put the script in 'bootstrap / command / git' and support the necessary functions. In addition, [git commit - git cl upload] in chromium and [git push origin branch - pull request
> In github web] are the same? ask politely.

Exactly. `git cl upload` is very similar to `git push origin branch` with a pull request.

The git custom command is very good apporach but it requires configuration in user side.
In other words, if someone wants to use the custom command, he/she will need to copy the script into GIT_EXEC directory first.
So, let's the custom command into bootstrap module. (as you already mentioned)
Then, we probably should implement the following commands at least.
- `./absolute pr #number`
- `./absolute pr upload`

**Comment by rockplani** [comment]
I was surprised at your detailed reply. @romandev 

The guesses about what you did not understand is correct.
It's also good to hear your thoughts about the issue.
(Specifically, the plan to include custom commands in the bootstrap module)

Thanks to your advice,
I think I will be able to proceed faster in solving this issue.
However, if I face a difficult problem in the future, I will post again.

**Comment by romandev** [comment]
@rockplani, you're welcome :)
We greately appreciate your contribution.
Happy coding!

**Comment by nadongguri** [comment]
I'm closing this issue, you can re-open it if you want, @rockplani and @GarlicB made similar feature before.
